skbio/sequence/_substitution.py

```diff
@@ -0,0 +1,518 @@
+        the identical scores, respectively, regardless of the character.
+
+        Parameters
+        ----------
+        alphabet : iterable
+            Characters that constitute the alphabet.
+        match : int or float
+            Score assigned to all matches.
+        mismatch : int or float
+            Score assigned to all mismatches.
+
+        Returns
+        -------
+        SubstitutionMatrix
+            Substitution matrix constructed from the alphabet and scores.
+
+        Examples
+        --------
+        >>> from skbio import SubstitutionMatrix
+        >>> mat = SubstitutionMatrix.identity('ACGT', 1, -2)
+        >>> mat.alphabet
+        ('A', 'C', 'G', 'T')
+        >>> mat.scores
+        array([[ 1., -2., -2., -2.],
+               [-2.,  1., -2., -2.],
+               [-2., -2.,  1., -2.],
+               [-2., -2., -2.,  1.]])
+
+        """
+        alphabet = tuple(alphabet)
+        scores = np.identity(len(alphabet)) * (match - mismatch) + mismatch
+        return cls(alphabet, scores)
+
+    @classonlymethod
+    @experimental(as_of='0.5.10')
+    def by_name(cls, name):
+        """Load a pre-defined substitution matrix by its name.
+
+        Parameters
+        ----------
+        name : str
+            Name of the substitution matrix.
+
+        Returns
+        -------
+        SubstitutionMatrix
+            Named substitution matrix.
+
+        Raises
+        ------
+        ValueError
+            If named substitution matrix does not exist.
+
+        See Also
+        --------
+        get_names
+
+        Notes
+        -----
+        Available substitution matrix names can be obtained by ``get_names``.
+        Currently, the following names are supported:
+
+        - `NUC.4.4` (a.k.a. DNAfull): A nucleotide substitution matrix covering
+          all definite and degenerate nucleotides.
+
+        - Point Accepted Mutation (PAM) [1]_: A set of amino acid substitution
+          matrices, including `PAM30`, `PAM70` and `PAM250`.
+
+        - BLOcks SUbstitution Matrix (BLOSUM) [2]_: A set of amino acid
+          substitution matrices, including `BLOSUM45`, `BLOSUM50`, `BLOSUM62`,
+          `BLOSUM80` and `BLOSUM90`.
+
+        References
+        ----------
+        .. [1] Dayhoff, M., Schwartz, R., & Orcutt, B. (1978). A model of
+           evolutionary change in proteins. Atlas of protein sequence and
+           structure, 5, 345-352.
+        .. [2] Henikoff, S., & Henikoff, J. G. (1992). Amino acid substitution
+           matrices from protein blocks. Proceedings of the National Academy of
+           Sciences, 89(22), 10915-10919.
+
+        Examples
+        --------
+        >>> from skbio import SubstitutionMatrix
+        >>> mat = SubstitutionMatrix.by_name('BLOSUM62')
+        >>> len(mat.alphabet)
+        24
+        >>> mat['M', 'K']
+        -1.0
+
+        """
+        try:
+            return named_substitution_matrices[name]
```

Comment:
Normalizing case would probably dampen some user frustration, e.g. `name.lower()` and ensuring `named_substitution_matrices` supports both lower, upper and native casing

Reply:
Re: Normalizing case: Good point! Let me edit.

Re: Global scope. I noted this. At present other global variables are not upper case (for example see [genetic codes](https://github.com/scikit-bio/scikit-bio/blob/8a72c20c6ad50f9cb6d44f5fcf87be5f9582de58/skbio/sequence/_genetic_code.py#L761)). Therefore I followed the convention. But I agree that using upper case for global variables is the way to go. How about we create a separate PR to change all such variables into upper case?

Reply:
The genetic code example links to a variable that is described as module private as it is prefixed with '_'. If the use of `named_substitution_matrices` is similarly intended to be module private then perhaps following that form is best. (Or we create an issue to refactor genetic codes)

Reply:
Added normalization. Now `blosum62` and `BLOSUM62` both work. Let me also add `_` before `named_substitution_matrices`.

Reply:
Looking at these changes now. The solution works (ie no concern if this is merged as is), though this would be more direct rather than a search. Alternatively, the keys in _named_substitition_matrices could be forced to a specific case and then the check can be readily normalized. 

```python
if name in _named_substitition_matrices:
    return _named_substitition_matrices[name]
elif name.upper() in _named_substitition_matrices:
    return _named_substitition_matrices[name.upper()]
elif name.lower() in _named_substitition_matrices:
    return _named_substitition_matrices[name.lower()]
else:
    raise KeyError(f"Unrecognized substititution matrix {name}")
```